process/sync/shardblock.go

```diff
@@ -108,42 +66,13 @@ func NewShardBootstrap(
 	base.requestMiniBlocks = boot.requestMiniBlocksFromHeaderWithNonceIfMissing
 	base.getHeaderFromPool = boot.getShardHeaderFromPool
 
-	//TODO: ResolversFinder should be replaced with RequestHandler after it would be refactored and RequestedItemsHandler
-	//should be then removed from ShardBootstrap
-
-	//there is one header topic so it is ok to save it
-	//TODO make use of requestHandler
-	hdrResolver, err := resolversFinder.CrossShardResolver(factory.ShardBlocksTopic, sharding.MetachainShardId)
-	if err != nil {
-		return nil, err
-	}
-
-	//sync should request the missing block body on the intrashard topic
-	miniBlocksResolver, err := resolversFinder.IntraShardResolver(factory.MiniBlocksTopic)
-	if err != nil {
-		return nil, err
-	}
-
 	//placed in struct fields for performance reasons
 	base.headerStore = boot.store.GetStorer(dataRetriever.BlockHeaderUnit)
 	hdrNonceHashDataUnit := dataRetriever.ShardHdrNonceHashDataUnit + dataRetriever.UnitType(boot.shardCoordinator.SelfId())
 	base.headerNonceHashStore = boot.store.GetStorer(hdrNonceHashDataUnit)
 
-	hdrRes, ok := hdrResolver.(dataRetriever.HeaderResolver)
-	if !ok {
-		return nil, process.ErrWrongTypeAssertion
-	}
-
-	base.hdrRes = hdrRes
 	base.forkInfo = process.NewForkInfo()
 
-	miniBlocksRes, ok := miniBlocksResolver.(dataRetriever.MiniBlocksResolver)
-	if !ok {
-		return nil, process.ErrWrongTypeAssertion
-	}
-
-	boot.miniBlocksResolver = miniBlocksRes
-
 	boot.chRcvHdrNonce = make(chan bool)
```

Comment:
maybe you can create a method in baseSync that init common code
boot.chRcvHdrNonce = make(chan bool)
	boot.chRcvHdrHash = make(chan bool)
	boot.chRcvMiniBlocks = make(chan bool)
boot.chStopSync = make(chan bool)

	boot.setRequestedHeaderNonce(nil)
	boot.setRequestedHeaderHash(nil)
	boot.setRequestedMiniBlocks(nil)

Reply:
Done